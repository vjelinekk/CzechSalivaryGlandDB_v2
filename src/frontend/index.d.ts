export {}

import {
    ipcAPIDeleteChannels,
    ipcAPIGetChannels,
    ipcAPISaveChannels,
    ipcExportChannels,
} from '../ipc/ipcChannels'
import { FormType } from './constants'
import {
    KaplanMeierData,
    PatientType,
    PlannedPatientsMap,
    Study,
    JSONObject,
} from './types'

declare global {
    interface Window {
        api: {
            save: (
                channel: ipcAPISaveChannels,
                data: JSON
            ) => Promise<number | null>
            updatePatientsStudies: (
                patientId: number,
                patientType: FormType,
                studies: Study[]
            ) => Promise<boolean>
            insert: (
                channel: ipcAPIInsertChannels,
                data: JSON
            ) => Promise<boolean>
            delete: (
                chanel: ipcAPIDeleteChannels,
                data: JSON
            ) => Promise<boolean>
            deletePatientFromStudy: (
                studyId: number,
                patientId: number,
                patientType: FormType
            ) => Promise<boolean>
            get: (
                channel: ipcAPIGetChannels,
                formType?: FormType
            ) => Promise<PatientType[]>
            getStudiesByFormType: (formType: FormType) => Promise<Study[]>
            getStudiesByPatientId: (
                id: number,
                patientType: FormType
            ) => Promise<Study[]>
            searchPatientsByNameSurnameRC: (
                search: string
            ) => Promise<PatientType[]>
            getFilteredPatients: (
                filter: FilteredColumns,
                studyId?: number
            ) => Promise<PatientType[]>
            getKaplanMeierData: (
                kaplanMeierType: KaplanMeierType,
                filter: FilteredColumns
            ) => Promise<KaplanMeierData>
            getPlannedPatientsBetweenDates: (
                startDate: Date,
                endDate: Date
            ) => Promise<PlannedPatientsMap>
        }
        export: {
            export: (
                channel: ipcExportChannels,
                patients: PatientType[]
            ) => Promise<void>
        }
        import: {
            import: () => Promise<void>
        }
        fs: {
            save: () => Promise<string | null>
            getFileIcon: (fileName: string) => Promise<string>
            getFileName: (filePath: string) => Promise<string>
            open: (filePath: string) => void
            loadJson: (filePath: string) => Promise<Record<string, string>>
        }
        encryption: {
            setEncryptionKey: (key: string) => void
            isPasswordSet: () => Promise<boolean | null>
            isEncryptionEnabled: () => Promise<boolean | null>
            insertPasswordRow: (
                password: string,
                usingEncryption: boolean
            ) => Promise<void>
            insertPassword: (password: string) => Promise<void>
            insertUsingEncryption: (enabled: boolean) => Promise<void>
            validatePassword: (password: string) => Promise<boolean>
            generateEncryptionKey: () => Promise<string>
        }
        backUp: {
            backUp: () => Promise<void>
            loadBackUp: () => Promise<void>
        }
    }
}
