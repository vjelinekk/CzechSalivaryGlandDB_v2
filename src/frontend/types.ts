import { Dispatch, SetStateAction } from 'react'
import { Components, FormStates, FormType, StudyType } from './constants'

export interface activeComponentState {
    component: Components
    activePatient?: PatientType | null
    studyType?: StudyType | null
    activeStudy?: Study | null
}

export interface EditSavedState {
    done: boolean
    saved: boolean | null
}

export interface GlandComponentProps {
    formData: PatientData | null
    setFormData: Dispatch<SetStateAction<PatientData | null>>
    setFormErrors?: Dispatch<SetStateAction<string[]>>
    disabled: boolean
}

export interface GlandFormProps {
    data?: PatientData
    defaultFormState: FormStates
    setActiveComponent?: Dispatch<SetStateAction<activeComponentState>>
    editSaved?: EditSavedState
    setEditSaved?: Dispatch<SetStateAction<EditSavedState>>
    setActivePatient?: Dispatch<SetStateAction<PatientType | null>>
    defaultSelectedStudies?: Study[]
    idStudie?: number
}

export interface PatientData {
    [key: string]: string | number | FormType
    id?: number
    form_type?: FormType
    attachments?: string
    jmeno?: string
    prijmeni?: string
    id_pacient?: string
    rodne_cislo?: string
    vek_pri_diagnoze?: number
    pohlavi?: string
    kraj?: string
    jine_nadorove_onemocneni_v_oa?: string
    specifikace_mista_vyskytu_jineho_karcinomu?: string
    jine_onemocneni_velkych_slinnych_zlaz_v_oa?: string
    specifikace_onemocneni?: string
    koureni?: string
    pocet_cigaret_denne?: number
    jak_dlouho_kouri?: number
    pocet_balickoroku?: number
    abusus_alkoholu?: string
    rok_diagnozy?: string
    diagnoza_mkn_10?: string
    strana_nalezu?: string
    diagnosticke_metody?: string
    fnab?: string
    vysledek_fnab?: string
    core_biopsie?: string
    core_vysledek?: string
    mukoepidermoidni_karcinom_core?: string
    adenoidne_cysticky_karcinom_core?: string
    polymorfni_adenokarcinom_core?: string
    intraduktalni_karcinom_core?: string
    salivarni_karcinom_nos_core?: string
    karcinom_z_pleomorfniho_adenomu_core?: string
    spatne_diferencovany_karcinom_core?: string
    otevrena_biopsie?: string
    otevrena_vysledek?: string
    mukoepidermoidni_karcinom_otevrena?: string
    adenoidne_cysticky_karcinom_otevrena?: string
    polymorfni_adenokarcinom_otevrena?: string
    intraduktalni_karcinom_otevrena?: string
    salivarni_karcinom_nos_otevrena?: string
    karcinom_z_pleomorfniho_adenomu_otevrena?: string
    spatne_diferencovany_karcinom_otevrena?: string
    datum_zahajeni_lecby?: string
    typ_terapie?: string
    rozsah_chirurgicke_lecby?: string
    blokova_krcni_disekce?: string
    strana_blokove_krcni_disekce?: string
    typ_nd?: string
    rozsah_nd?: string
    adjuvantni_terapie?: string
    typ_nechirurgicke_terapie?: string
    histopatologie_vysledek?: string
    mukoepidermoidni_karcinom_histopatologie?: string
    adenoidne_cysticky_karcinom_histopatologie?: string
    polymorfni_adenokarcinom_histopatologie?: string
    intraduktalni_karcinom_histopatologie?: string
    salivarni_karcinom_nos_histopatologie?: string
    karcinom_z_pleomorfniho_adenomu_histopatologie?: string
    spatne_diferencovany_karcinom_histopatologie?: string
    velikost_nadoru_histopatologie?: number
    velikost_nadoru_neurcena_histopatologie?: string
    okraj_resekce_histopatologie?: string
    lymfovaskularni_invaze_histopatologie?: string
    perineuralni_invaze_histopatologie?: string
    pocet_lymfatickych_uzlin_s_metastazou_histopatologie?: string
    extranodalni_sireni_histopatologie?: string
    extranodalni_sireni_vysledek_histopatologie?: string
    prokazane_vzdalene_metastazy_histopatologie?: string
    misto_vyskytu_vzdalene_metastazy_histopatologie?: string
    t_klasifikace_klinicka?: string
    n_klasifikace_klinicka?: string
    m_klasifikace_klinicka?: string
    tnm_klasifikace_klinicka?: string
    t_klasifikace_patologicka?: string
    n_klasifikace_patologicka?: string
    m_klasifikace_patologicka?: string
    tnm_klasifikace_patologicka?: string
    datum_prvni_kontroly_po_lecbe?: string
    perzistence?: string
    datum_prokazani_perzistence?: string
    recidiva?: string
    datum_prokazani_recidivy?: string
    stav?: string
    datum_umrti?: string
    posledni_kontrola?: string
    planovana_kontrola?: string
    poznamky?: string
}

export interface ParotidPatientData extends PatientData {
    funkce_n_vii_dle_h_b_predoperacne?: string
    funkce_n_vii_dle_h_b_pooperacne?: string
    pooperacni_komplikace?: string
    jine_pooperacni_komplikace?: string
}

export interface SubmandibularPatientData extends PatientData {
    funkce_n_vii_dle_h_b_predoperacne?: string
    funkce_n_vii_dle_h_b_pooperacne?: string
}

export interface Study {
    id?: number
    nazev_studie?: string
    typ_studie?: StudyType
}

export interface PatientInStudy {
    id?: number
    id_pacient_db?: number
    id_studie?: number
    typ_pacienta?: FormType
}

export type SublingualPatientData = PatientData

export type PatientType =
    | ParotidPatientData
    | SubmandibularPatientData
    | SublingualPatientData

export interface FilteredColumns {
    [key: string]: string[] | FormType[]
    form_type: FormType[]
    typ_terapie: string[]
    histopatologie_vysledek: string[]
}
