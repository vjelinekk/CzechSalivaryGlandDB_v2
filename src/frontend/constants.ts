export enum Components {
    patientsList = 'patients-list',
    addPatient = 'add-patient',
    studiesList = 'studies-list',
    addStudy = 'add-study',
    parotidGlandForm = 'parotid-gland-form',
    sublingualGlandForm = 'sublingual-gland-form',
    submandibularGlandForm = 'submandibular-gland-form',
    study = 'study',
}

export enum StudyTypes {
    submandibular = 1,
    sublingual = 2,
    parotid = 3,
    special = 4,
}

export enum FormStates {
    add = 'add',
    edit = 'edit',
    view = 'view',
}

export const formTypeToStringMap: { [key: number]: string } = {
    1: 'podčelistní',
    2: 'podjazyková',
    3: 'příušní',
}

export const formTypeToBasicStringMap: { [key: number]: string } = {
    1: 'podcelistni',
    2: 'podjazykova',
    3: 'priusni',
}

export const studyTypeToStringMap: { [key: number]: string } = {
    1: 'podčelistní',
    2: 'podjazyková',
    3: 'příušní',
    4: 'speciální',
}

export enum dbLabels {
    jmeno = 'jmeno',
    prijmeni = 'prijmeni',
    id_pacient = 'id_pacient',
    rodne_cislo = 'rodne_cislo',
    vek_pri_diagnoze = 'vek_pri_diagnoze',
    pohlavi = 'pohlavi',
    kraj = 'kraj',
    jine_nadorove_onemocneni_v_oa = 'jine_nadorove_onemocneni_v_oa',
    specifikace_mista_vyskytu_jineho_karcinomu = 'specifikace_mista_vyskytu_jineho_karcinomu',
    jine_onemocneni_velkych_slinnych_zlaz_v_oa = 'jine_onemocneni_velkych_slinnych_zlaz_v_oa',
    specifikace_onemocneni = 'specifikace_onemocneni',
    koureni = 'koureni',
    pocet_cigaret_denne = 'pocet_cigaret_denne',
    jak_dlouho_kouri = 'jak_dlouho_kouri',
    pocet_balickoroku = 'pocet_balickoroku',
    abusus_alkoholu = 'abusus_alkoholu',
    rok_diagnozy = 'rok_diagnozy',
    diagnoza_mkn_10 = 'diagnoza_mkn_10',
    strana_nalezu = 'strana_nalezu',
    funkce_n_vii_dle_h_b_predoperacne = 'funkce_n_vii_dle_h_b_predoperacne',
    diagnosticke_metody = 'diagnosticke_metody',
    fnab = 'fnab',
    vysledek_fnab = 'vysledek_fnab',
    core_biopsie = 'core_biopsie',
    core_vysledek = 'core_vysledek',
    mukoepidermoidni_karcinom_core = 'mukoepidermoidni_karcinom_core',
    adenoidne_cysticky_karcinom_core = 'adenoidne_cysticky_karcinom_core',
    polymorfni_adenokarcinom_core = 'polymorfni_adenokarcinom_core',
    intraduktalni_karcinom_core = 'intraduktalni_karcinom_core',
    salivarni_karcinom_nos_core = 'salivarni_karcinom_nos_core',
    karcinom_z_pleomorfniho_adenomu_core = 'karcinom_z_pleomorfniho_adenomu_core',
    spatne_diferencovany_karcinom_core = 'spatne_diferencovany_karcinom_core',
    otevrena_biopsie = 'otevrena_biopsie',
    otevrena_vysledek = 'otevrena_vysledek',
    mukoepidermoidni_karcinom_otevrena = 'mukoepidermoidni_karcinom_otevrena',
    adenoidne_cysticky_karcinom_otevrena = 'adenoidne_cysticky_karcinom_otevrena',
    polymorfni_adenokarcinom_otevrena = 'polymorfni_adenokarcinom_otevrena',
    intraduktalni_karcinom_otevrena = 'intraduktalni_karcinom_otevrena',
    salivarni_karcinom_nos_otevrena = 'salivarni_karcinom_nos_otevrena',
    karcinom_z_pleomorfniho_adenomu_otevrena = 'karcinom_z_pleomorfniho_adenomu_otevrena',
    spatne_diferencovany_karcinom_otevrena = 'spatne_diferencovany_karcinom_otevrena',
    datum_zahajeni_lecby = 'datum_zahajeni_lecby',
    typ_terapie = 'typ_terapie',
    rozsah_chirurgicke_lecby = 'rozsah_chirurgicke_lecby',
    blokova_krcni_disekce = 'blokova_krcni_disekce',
    strana_blokove_krcni_disekce = 'strana_blokove_krcni_disekce',
    typ_nd = 'typ_nd',
    rozsah_nd = 'rozsah_nd',
    funkce_n_vii_dle_h_b_pooperacne = 'funkce_n_vii_dle_h_b_pooperacne',
    pooperacni_komplikace = 'pooperacni_komplikace',
    jine_pooperacni_komplikace = 'jine_pooperacni_komplikace',
    adjuvantni_terapie = 'adjuvantni_terapie',
    typ_nechirurgicke_terapie = 'typ_nechirurgicke_terapie',
    histopatologie_vysledek = 'histopatologie_vysledek',
    mukoepidermoidni_karcinom_histopatologie = 'mukoepidermoidni_karcinom_histopatologie',
    adenoidne_cysticky_karcinom_histopatologie = 'adenoidne_cysticky_karcinom_histopatologie',
    polymorfni_adenokarcinom_histopatologie = 'polymorfni_adenokarcinom_histopatologie',
    intraduktalni_karcinom_histopatologie = 'intraduktalni_karcinom_histopatologie',
    salivarni_karcinom_nos_histopatologie = 'salivarni_karcinom_nos_histopatologie',
    karcinom_z_pleomorfniho_adenomu_histopatologie = 'karcinom_z_pleomorfniho_adenomu_histopatologie',
    spatne_diferencovany_karcinom_histopatologie = 'spatne_diferencovany_karcinom_histopatologie',
    velikost_nadoru_histopatologie = 'velikost_nadoru_histopatologie',
    velikost_nadoru_neurcena_histopatologie = 'velikost_nadoru_neurcena_histopatologie',
    okraj_resekce_histopatologie = 'okraj_resekce_histopatologie',
    lymfovaskularni_invaze_histopatologie = 'lymfovaskularni_invaze_histopatologie',
    perineuralni_invaze_histopatologie = 'perineuralni_invaze_histopatologie',
    pocet_lymfatickych_uzlin_s_metastazou_histopatologie = 'pocet_lymfatickych_uzlin_s_metastazou_histopatologie',
    extranodalni_sireni_histopatologie = 'extranodalni_sireni_histopatologie',
    extranodalni_sireni_vysledek_histopatologie = 'extranodalni_sireni_vysledek_histopatologie',
    prokazane_vzdalene_metastazy_histopatologie = 'prokazane_vzdalene_metastazy_histopatologie',
    misto_vyskytu_vzdalene_metastazy_histopatologie = 'misto_vyskytu_vzdalene_metastazy_histopatologie',
    t_klasifikace_klinicka = 't_klasifikace_klinicka',
    n_klasifikace_klinicka = 'n_klasifikace_klinicka',
    m_klasifikace_klinicka = 'm_klasifikace_klinicka',
    tnm_klasifikace_klinicka = 'tnm_klasifikace_klinicka',
    t_klasifikace_patologicka = 't_klasifikace_patologicka',
    n_klasifikace_patologicka = 'n_klasifikace_patologicka',
    m_klasifikace_patologicka = 'm_klasifikace_patologicka',
    tnm_klasifikace_patologicka = 'tnm_klasifikace_patologicka',
    datum_prvni_kontroly_po_lecbe = 'datum_prvni_kontroly_po_lecbe',
    perzistence = 'perzistence',
    datum_prokazani_perzistence = 'datum_prokazani_perzistence',
    recidiva = 'recidiva',
    datum_prokazani_recidivy = 'datum_prokazani_recidivy',
    stav = 'stav',
    datum_umrti = 'datum_umrti',
    posledni_kontrola = 'posledni_kontrola',
    planovana_kontrola = 'planovana_kontrola',
    poznamky = 'poznamky',
}

export enum TNMClassificationResult {
    stageI = 'Stage I',
    stageII = 'Stage II',
    stageIII = 'Stage III',
    stageIVA = 'Stage IVA',
    stageIVB = 'Stage IVB',
    stageIVC = 'Stage IVC',
}

export enum FormType {
    podcelistni = 1,
    podjazykove = 2,
    priusni = 3,
}

export const studyTypeToFormTypeMap: { [key: number]: FormType } = {
    1: FormType.podcelistni,
    2: FormType.podjazykove,
    3: FormType.priusni,
}
