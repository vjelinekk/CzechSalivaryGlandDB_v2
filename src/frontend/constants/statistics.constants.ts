import { InferenceChiSquareCategories } from '../enums/statistics.enums'

export const COLORS = [
    '#0088FE',
    '#00C49F',
    '#FFBB28',
    '#FF8042',
    '#9c27b0',
    '#e91e63',
    '#3f51b5',
]

export const CATEGORY_TITLES: Record<InferenceChiSquareCategories, string> = {
    [InferenceChiSquareCategories.histologicalTypes]:
        'Histologický typ nádoru - možnosti k výběru',
    [InferenceChiSquareCategories.tClassification]:
        'T klasifikace - možnosti k výběru',
    [InferenceChiSquareCategories.nClassification]:
        'N klasifikace - možnosti k výběru',
    [InferenceChiSquareCategories.mClassification]:
        'M klasifikace - možnosti k výběru',
    [InferenceChiSquareCategories.persistence]:
        'Perzistence - možnosti k výběru',
    [InferenceChiSquareCategories.recurrence]: 'Recidiva - možnosti k výběru',
    [InferenceChiSquareCategories.state]: 'Stav pacienta - možnosti k výběru',
}

export const TAB_TITLES: Record<InferenceChiSquareCategories, string> = {
    [InferenceChiSquareCategories.histologicalTypes]: 'Histologický typ',
    [InferenceChiSquareCategories.tClassification]: 'T klasifikace',
    [InferenceChiSquareCategories.nClassification]: 'N klasifikace',
    [InferenceChiSquareCategories.mClassification]: 'M klasifikace',
    [InferenceChiSquareCategories.persistence]: 'Perzistence',
    [InferenceChiSquareCategories.recurrence]: 'Recidiva',
    [InferenceChiSquareCategories.state]: 'Stav pacienta',
}
