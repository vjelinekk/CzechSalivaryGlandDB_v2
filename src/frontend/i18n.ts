import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'

const loadTranslations = async () => {
    console.log('Loading translations...')

    try {
        const enTranslation = await window.fs.loadJson(
            'public/locales/translation-en.json'
        )
        const csTranslation = await window.fs.loadJson(
            'public/locales/translation-cs.json'
        )
        const skTranslation = await window.fs.loadJson(
            'public/locales/translation-sk.json'
        )
        // const deTranslation = await window.fs.loadJson('public/locales/translation-de.json');
        // const esTranslation = await window.fs.loadJson('public/locales/translation-es.json');

        const csFormTranslation = await window.fs.loadJson(
            'public/locales/form-translation-cs.json'
        )
        const enFormTranslation = await window.fs.loadJson(
            'public/locales/form-translation-en.json'
        )
        const skFormTranslation = await window.fs.loadJson(
            'public/locales/form-translation-sk.json'
        )
        // const deFormTranslation = await window.fs.loadJson('public/locales/form-translation-de.json');
        // const esFormTranslation = await window.fs.loadJson('public/locales/form-translation-es.json');

        return {
            en: { translation: { ...enTranslation, ...enFormTranslation } },
            cs: { translation: { ...csTranslation, ...csFormTranslation } },
            sk: { translation: { ...skTranslation, ...skFormTranslation } },
            // de: { translation: { ...deTranslation, ...deFormTranslation} },
            // es: { translation: { ...esTranslation, ...esFormTranslation} },
        }
    } catch (error) {
        console.error('Error loading translations:', error.message)
        return {}
    }
}

export const initI18n = async () => {
    const resources = await loadTranslations()
    // eslint-disable-next-line import/no-named-as-default-member
    i18n.use(initReactI18next).init({
        resources,
        lng: 'cs',
        fallbackLng: 'en',
    })
}

export default i18n
