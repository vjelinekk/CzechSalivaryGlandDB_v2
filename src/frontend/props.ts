import { Dispatch, SetStateAction } from 'react'
import { PatientData, PatientType } from './types'

export interface CommonFormInputProps {
    dbLabel: string
    data: string | number | Array<string>
    setFormData: Dispatch<SetStateAction<PatientType> | null>
    disabled: boolean
}

export interface CommonFormProps {
    data: PatientData
}
