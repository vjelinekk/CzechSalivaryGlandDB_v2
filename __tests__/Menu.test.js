/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import Menu from '../src/frontend/components/menu'
import { components } from '../src/frontend/constants'
import { test, expect, jest, afterEach } from '@jest/globals'

// Mock setActiveComponent function
const mockSetActiveComponent = jest.fn()

afterEach(cleanup)

test('Menu renders correctly', () => {
    const { getByText } = render(
        <Menu setActiveComponent={mockSetActiveComponent} />
    )

    // Find buttons by their text content
    const listPatientButton = getByText('Seznam pacientů')
    const addPatientButton = getByText('Přidat pacienta')
    const studiesButton = getByText('Studie')
    const addStudyButton = getByText('Přidat studii')

    // Simulate button clicks
    fireEvent.click(listPatientButton)
    fireEvent.click(addPatientButton)
    fireEvent.click(studiesButton)
    fireEvent.click(addStudyButton)

    // Check if setActiveComponent was called with the correct arguments
    expect(mockSetActiveComponent).toHaveBeenCalledWith(components.patientsList)
    expect(mockSetActiveComponent).toHaveBeenCalledWith(components.addPatient)
    expect(mockSetActiveComponent).toHaveBeenCalledWith(components.studiesList)
    expect(mockSetActiveComponent).toHaveBeenCalledWith(components.addStudy)
})
