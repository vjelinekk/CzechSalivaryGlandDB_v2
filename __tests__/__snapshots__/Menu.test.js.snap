// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Menu renders correctly 1`] = `
<div
  id="sidebar"
>
  <div
    id="bg"
  />
  <div
    className="welcome"
  >
    <img
      alt="Česká společnost otorinolaryngologie a chirurgie hlavy a krku"
      id="logo"
      src="../img/LOGO_splt.png"
    />
  </div>
  <ul>
    <li>
      <button
        id="list-patient"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="pacienti"
          src="../img/pacienti_outline.png"
        />
        Seznam pacientů
      </button>
    </li>
    <li>
      <button
        id="add-patient"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="pacient_pridat"
          src="../img/pacient_pridat_outline.png"
        />
        Přidat pacienta
      </button>
    </li>
    <li>
      <button
        id="studies-btn"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="studie"
          src="../img/studie_outline.png"
        />
        Studie
      </button>
    </li>
    <li>
      <button
        id="add-study"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="studie_pridat"
          src="../img/studie_pridat_outline.png"
        />
        Přidat studii
      </button>
    </li>
  </ul>
</div>
`;

exports[`Menu renders correctly 1`] = `
<div
  id="sidebar"
>
  <div
    id="bg"
  />
  <div
    className="welcome"
  >
    <img
      alt="Česká společnost otorinolaryngologie a chirurgie hlavy a krku"
      id="logo"
      src="../img/LOGO_splt.png"
    />
  </div>
  <ul>
    <li>
      <button
        id="list-patient"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="pacienti"
          src="../img/pacienti_outline.png"
        />
        Seznam pacientů
      </button>
    </li>
    <li>
      <button
        id="add-patient"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="pacient_pridat"
          src="../img/pacient_pridat_outline.png"
        />
        Přidat pacienta
      </button>
    </li>
    <li>
      <button
        id="studies-btn"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="studie"
          src="../img/studie_outline.png"
        />
        Studie
      </button>
    </li>
    <li>
      <button
        id="add-study"
        onClick={[Function]}
      >
        <img
          className="icon"
          id="studie_pridat"
          src="../img/studie_pridat_outline.png"
        />
        Přidat studii
      </button>
    </li>
  </ul>
</div>
`;
